wallet/src/main/java/com/android/identity/issuance/WalletDocumentMetadata.kt

```diff
@@ -0,0 +1,164 @@
+package com.android.identity.issuance
+
+import com.android.identity.cbor.annotation.CborSerializable
+import com.android.identity.document.DocumentMetadata
+import com.android.identity.issuance.remote.WalletServerProvider
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlinx.io.bytestring.ByteString
+import kotlinx.io.bytestring.isEmpty
+import kotlin.concurrent.Volatile
+
+class WalletDocumentMetadata private constructor(
+    serializedData: ByteString?,
+    private val saveFn: suspend (data: ByteString) -> Unit
+) : DocumentMetadata {
```

Comment:
Very nice example of how a wallet app can specialize DocumentMetadata to include e.g. information about the origin of the document (issuer metadata). Cool.